src/components/exercises/App1/redux/slices/userSlice.js

```diff
@@ -1,21 +1,70 @@
-import { createSlice} from "@reduxjs/toolkit";
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+
+export const logInThunk = createAsyncThunk(
+  "loggedInUser/logInThunk",
+  async () => {
+    const response = await fetch("login", { method: "POST" }).then((response) =>
+      response.json()
+    );
+    return response;
+  }
+);
+export const logOutThunk = createAsyncThunk(
+  "loggedInUser/logOutThunk",
+  async () => {
+    const response = await fetch("logout", { method: "POST" }).then(
+      (response) => response.json()
+    );
+    return response;
+  }
+);
+export const logUserInThunk = createAsyncThunk(
+  "loggedInUser/logInUserThunk",
+  async () => {
+    const response = await fetch("getlogin-User").then((response) =>
+      response.json()
+    );
+    return response;
+  }
+);
 
 export const userSlice = createSlice({
-    name: 'user',
-    initialState: {},
-    reducers: {
-        log_in: (state, action) => {
-            state = action.payload.user;
-        },
-        log_out: (state, action) => {
-            return null;
-        },
-        get_user: (state, action) => {
-            state = action.payload.user;
-        }
-    }
+  name: "user",
+  initialState: {
+    status: "idle",
+    loggedInUser: null,
+  },
+  reducers: {
+    get_user: (state, action) => {
+      state.loggedInUser = action.payload;
+    },
+  },
+  extraReducers(builder) {
+    builder.addCase(logInThunk.fulfilled, (state, action) => {
+      state.loggedInUser = action.payload.user;
```

Comment:
Add check if it was logged in successfully, if success === true set loggedInUser